recipes/occt/all/conanfile.py

```diff
@@ -0,0 +1,108 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class OcctConan(ConanFile):
+    name = "occt"
+    description = """a software development platform providing services for 3D
+        surface and solid modeling, CAD data exchange, and visualization."""
+    homepage = "https://github.com/Open-Cascade-SAS/OCCT"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "LGPL-2.1-or-later"
+    topics = ("3D", "modeling")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    generators = "cmake"
+    requires = [("tcl/8.6.10"),
+                ("tk/8.6.10"),
+                ("freetype/2.10.2")]
+    _source_subfolder = "src"
+    _cmake = None
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name.upper() + "-" + \
+            self.version.replace(".", "_")
+        tools.rename(extracted_dir, self._source_subfolder)
+
+    def _patch_sources(self):
+        tools.replace_in_file(
+            os.path.join(self._source_subfolder, "CMakeLists.txt"),
+            "project (OCCT)",
+            '''project (OCCT)
+                include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+                conan_basic_setup()''')
+        tools.replace_in_file(
+            os.path.join(self._source_subfolder, "adm/cmake/tk.cmake"),
+            "${CSF_TclTkLibs}",
+            self.deps_cpp_info["tk"].libs[0])
+        tools.replace_in_file(
+            os.path.join(self._source_subfolder, "adm/cmake/tcl.cmake"),
+            "${CSF_TclLibs}",
+            self.deps_cpp_info["tcl"].libs[0])
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["3RDPARTY_TCL_LIBRARY_DIR"] = \
+            self.deps_cpp_info["tcl"].lib_paths[0]
+        self._cmake.definitions["3RDPARTY_TCL_INCLUDE_DIR"] = \
+            self.deps_cpp_info["tcl"].include_paths[0]
+        self._cmake.definitions["3RDPARTY_TK_LIBRARY_DIR"] = \
+            self.deps_cpp_info["tk"].lib_paths[0]
+        self._cmake.definitions["3RDPARTY_TK_INCLUDE_DIR"] = \
+            self.deps_cpp_info["tk"].include_paths[0]
+        if not self.options.shared:
+            self._cmake.definitions["BUILD_LIBRARY_TYPE"] = "Static"
+
+        self._cmake.definitions["INSTALL_DIR_BIN"] = "bin"
+        self._cmake.definitions["INSTALL_DIR_INCLUDE"] = "include"
+        self._cmake.definitions["INSTALL_DIR_LIB"] = "lib"
+        self._cmake.definitions["INSTALL_DIR_RESOURCE"] = "res"
+
+        self._cmake.configure(source_folder=self._source_subfolder)
```

Comment:
this should also include the build folder

https://github.com/conan-io/conan-center-index/blob/689e2663fe41689259f61a5dbd3fe245e24c03b9/recipes/libdwarf/all/conanfile.py#L46

Reply:
It seems that with the cmake wrapper being used, the source_folder need not to be included at all.